client/js/store/modules/auth.js

```diff
@@ -59,7 +63,49 @@ export default {
     async requestSignup(context, { username, password }) {
       return handleLogin(context, username, password, true);
     },
-
+    async submitEmail(context, { username, email }) {
+      try {
+        const response = await fetch(`/user/submitEmail`, {
+          method: 'POST',
+          headers: new Headers({
+            'Content-Type': 'application/json',
+          }),
+          body: JSON.stringify({
+            username,
+            email,
+          }),
+          credentials: 'include',
+        });
+        const data = await response.json();
+        if (response.status !== 200) {
+          throw data.message;
+        }
+        context.commit('setEmail', email);
+      } catch (err) {
+        throw new Error(err);
+      }
+    },
+    async findEmail(context, { username }) {
+      try {
+        const response = await fetch(`/user/findEmail`, {
+          method: 'POST',
+          headers: new Headers({
+            'Content-Type': 'application/json',
+          }),
+          body: JSON.stringify({
+            username,
+          }),
+          credentials: 'include',
+        });
+        const data = await response.json();
+        if (response.status !== 200) {
+          throw data.message;
+        }
+        context.commit('setEmail', data);
+      } catch (err) {
+        throw new Error(err);
+      }
+    },
```

Comment:
Is this used to get the user email when loading the page? Isn't the `email` available on the User object itself in the `auth` module?

Reply:
I think we can directly grab the `email` off of the `User` object when logging in, refreshing auth status etc and committing it to the store